auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -674,8 +675,8 @@ private MethodSpec createReadMethod(ClassName className,
           }
         }
 
-        readMethod.addStatement("$T $N = $T.$L", builderField.get().type, builderField.get(),
-            autoValueClassName, builderMethod);
+        readMethod.addStatement("$T $N = $T.$N()", builderField.get().type, builderField.get(),
+            autoValueClassName, builderMethod.getSimpleName());
```

Comment:
`autoValueClassName` this is now a `ClassName` instance rather than a `TypeName` instance. We needed a `ClassName` instance otherwise this would have generated `Foo.Builder<T> builder = Foo<T>.builder();`